pages/admin/dashboard/modules.tsx

```diff
@@ -58,20 +48,36 @@ const ModulesPage: Page = () => {
                 </Button>
                 <Button variant="link">Published</Button>
             </Flex>
-            <SimpleGrid minChildWidth="20rem" spacing={10}>
-                {modules.map(
-                    ({ _id, title, toolID, lastUpdated, createdBy }) => (
-                        <ModuleCard
-                            key={_id}
-                            moduleId={_id}
-                            title={title}
-                            tool={toolID}
-                            lastUpdated={lastUpdated}
-                            author={createdBy.join(", ")}
-                        />
-                    ),
-                )}
-            </SimpleGrid>
+        </>
+    );
+};
+
+const Modules: React.FC = () => {
+    const { data, error } = useSWR("/api/module/getAll", fetcher);
+    if (error) return "An error has occurred.";
+    if (!data) return <Spinner color="brand.lime" size="xl" />;
+
+    return (
+        <SimpleGrid minChildWidth="20rem" spacing={10}>
+            {data.map(({ _id, title, toolID, lastUpdated, createdBy }) => (
```

Comment:
here we map directly from data - should we set data to a separate var with `useState`

Reply:
I don't think it's necessary. If we don't need to mutate the data we can keep it as is.